src/runtime_src/core/pcie/windows/device_windows.h

```diff
@@ -43,7 +47,10 @@ class device_windows : public device_pcie {
     virtual void update_shell(uint64_t _deviceID, std::string flashType, std::string& primary, std::string& secondary) const;
     virtual void update_SC(uint64_t _deviceID, std::string& file) const;
     //end flash functions
-
+    virtual void get_ip_layout(uint64_t _deviceID, struct ip_layout **ipLayout, DWORD size) const;
+    virtual DWORD get_ip_layoutsize(uint64_t _deviceID) const;
```

Comment:
Please no windows specific types in this interface, keep windows type isolated to implementation that uses windows specific APIs.